impress-org/givewp
# Milestone now displays aggregate revenue (based on selected forms)

_**Note: This PR is depended upon work in #5224 - it cannot be merged until that PR is first approved and merged! Additionally - this PR includes changes that impact block rendering logic in a handful of other PRs. I recommend waiting until all open PRs related to Milestones are first merged!**_

Resolves #5228 

## Description
This PR introduces the Milestone model class, and with it, functionality to display aggregated revenue based on the forms selected for the Milestone block. Notably, with the new Milestone model, frontend rendering logic for Milestones has been moved out of the Block class and into the Milestone model. This allows us to build a milestone component template that is unaware of its context - be that in a shortcode or a block.

## Affects
This PR affects frontend rendering logic for Milestones, and the Milestone block class.

## Visuals
![AggregateRevenue](https://user-images.githubusercontent.com/5186078/92136573-2ce3c700-edda-11ea-825c-23643230c9b9.gif)

## Pre-review Checklist

## Testing Instructions
1. Add a new post or page
2. Add a Milestone block
3. Select forms to be used with the block
4. Check that revenue updates based on the total earnings of the selected forms
5. Save, and check that these changes are reflected on the frontend
